lib/grape/middleware/formatter.rb

```diff
@@ -81,6 +81,9 @@ def read_rack_input(body)
                 end
                 env['rack.request.form_input'] = env['rack.input']
               end
+            rescue MultiJson::ParseError
```

Comment:
This is specific to the JSON parser, so it should be in it, not here. Then you can raise a `Grape::Exception::InvalidMessageBody` error and re-rescue and just re-raise anything that is `Grape::Exception::Base` here. 

You might want to look at other parsers while you're at it, maybe the XML one has a similar problem and can be fixed in the same way?